File: aspnet/web-forms/overview/older-versions-getting-started/master-pages/specifying-the-title-meta-tags-and-other-html-headers-in-the-master-page-cs/samples/sample12.cs
protected override void OnLoadComplete(EventArgs e)
{
	// Set the page's title, if necessary
	if (string.IsNullOrEmpty(Page.Title) || Page.Title == "Untitled Page")
	{
		// Is this page defined in the site map?
		string newTitle = null;

		SiteMapNode current = SiteMap.CurrentNode;
		if (current != null)
		{
			newTitle = current.Title;
		}
		else
		{
			// Determine the filename for this page
			newTitle = System.IO.Path.GetFileNameWithoutExtension(Request.PhysicalPath);
		}

		Page.Title = newTitle;
	}

	base.OnLoadComplete(e);
}

File: aspnet/web-forms/overview/older-versions-getting-started/master-pages/specifying-the-title-meta-tags-and-other-html-headers-in-the-master-page-vb/samples/sample12.vb
Protected Overrides Sub OnLoadComplete(ByVal e As EventArgs)
 ' Set the page's title, if necessary
 If String.IsNullOrEmpty(Page.Title) OrElse Page.Title = "Untitled Page" Then
 ' Is this page defined in the site map?
 Dim newTitle As String = Nothing

 Dim current As SiteMapNode = SiteMap.CurrentNode
 If current IsNot Nothing Then
 newTitle = current.Title
 Else
 ' Determine the filename for this page
 newTitle = System.IO.Path.GetFileNameWithoutExtension(Request.PhysicalPath)
 End If

 Page.Title = newTitle
 End If

 MyBase.OnLoadComplete(e)
End Sub